Update Proxy Server configuration

evan-m-jackson opened the pull request.

evan-m-jackson committed:
Light Refactor of Code

- Delete 'HTTPServer.service' file
- Refactor 'Tar up the build step' in azure-vm.yml file
- Remove excess whitespace in IntegrationTest.cs file

evan-m-jackson committed:
Light Refactor of Code

- Delete 'HTTPServer.service' file
- Refactor 'Tar up the build step' in azure-vm.yml file
- Remove excess whitespace in IntegrationTest.cs file

evan-m-jackson committed:
Light Refactor of Code

- Delete 'HTTPServer.service' file
- Refactor 'Tar up the build step' in azure-vm.yml file
- Remove excess whitespace in IntegrationTest.cs file
- Format ResponseByPathTests.cs and ResponseTests.cs file

evan-m-jackson committed:
Light Refactor of Code

- Delete 'HTTPServer.service' file
- Refactor 'Tar up the build step' in azure-vm.yml file
- Remove excess whitespace in IntegrationTest.cs file
- Format ResponseByPathTests.cs and ResponseTests.cs file

evan-m-jackson committed:
Refactor File Names and Paths

- Create new directories and renamed existing directories
- Organize project files in sub-directories
- Refactor tests to correspond with specific classes
- Organize test files in sub-directories
- Perform heavy formatting
- Move test doubles into test directories

evan-m-jackson committed:
Refactor File Names and Paths

- Create new directories and renamed existing directories
- Organize project files in sub-directories
- Refactor tests to correspond with specific classes
- Organize test files in sub-directories
- Perform heavy formatting
- Move test doubles into test directories

evan-m-jackson committed:
Build Proxy Server Capabilities

- Add 'GetCode' method to 'Header' class to return status code
- Add if/else statement in Program class to allow for proxy server capabilities
- Add 'ProxyResponse' class and unit tests

evan-m-jackson committed:
Build Proxy Server Capabilities

- Add 'GetCode' method to 'Header' class to return status code
- Add if/else statement in Program class to allow for proxy server capabilities
- Add 'ProxyResponse' class and unit tests

evan-m-jackson committed:
Add PUT and DELETE capabilities to 'ProxyResponse' class

evan-m-jackson committed:
Add PUT and DELETE capabilities to 'ProxyResponse' class

evan-m-jackson committed:
Refactor ProxyResponse class and tests

- Refactor GetBody method
- Remove unnecessary test

evan-m-jackson committed:
Refactor ProxyResponse class and tests

- Refactor GetBody method
- Remove unnecessary test

paytonrules commented:
Commit messages here are much better - clearly state what's going on, make it much easier review the code as well (as I can see you did or didn't do what you meant). 

The next step is to make sure you've got a good summary and description in the PR. Many teams will have rules and guidelines for that, so it's good to develop that habit now.  Some example styles are here:

https://axolo.co/blog/p/part-3-github-pull-request-template

You don't need to use one of them but it's an example of what clients will be looking for when you get to that phase.

paytonrules commented on HTTPServerProject.Tests/ResponseTests.cs:
```diff
@@ -22,7 +22,7 @@ public void AddAndGetHeaderTest()
         var expected = new List<string>() { "HTTP/1.1 200 OK", "Location: Earth", "", "Hello World!" };
         var writer = new TestWriteStreams(eStream);
-        var headers = new List<string>(){"Location: Earth"};
+        var headers = new List<string>() { "Location: Earth" };
```
The commit here doesn't seem to match the actual code. Any idea what's going on here?

paytonrules replied:
Ah I see what happened - some commit message got re-used by mistake. This often happens in IntellJ because it has a (annoying IMO) feature of keeping the commit message from the last time you committed in the dialog. Make sure you double check commit messages when you commit.

evan-m-jackson replied:
Yes, my mistake. That was one of my first commits using Rider and I double tapped the commit message.

evan-m-jackson committed:
Rename namespaces to match file paths

evan-m-jackson committed:
Rename namespaces to match file paths

paytonrules commented on HTTPServerProject.Tests/HTTPServerProxyTests/TestProxyClient.cs:
```diff
@@ -0,0 +1,16 @@
+
+namespace HTTPServerProject.Tests.Proxy.Client;
+
+public class TestProxyClient : IProxyClient
```
It's not in this change - but what's the point of the empty interface `IProxyClient`?

evan-m-jackson replied:
It was just a way to make a test double for the `TcpClient` class, which I wrapped in the `ProxyClient` class

paytonrules replied:
But what's being doubled? There's no methods on IProxyClient so there's nothing to double.

evan-m-jackson replied:
I would have put the `GetStream()` method in IProxyClient but the `ProxyClient` returns a `NetworkStream` and the `TestProxyClient` returns a `List<string>`.  Wouldn't that cause an error in the interface?

paytonrules commented on HTTPServerProject.Tests/HTTPServerReadTests/BodyTests.cs:
```diff
@@ -0,0 +1,25 @@
+        var reader = new TestReadStreams(request);
+
+        var header = new Header(reader);
+        var initialLine = header.GetLine();
```
I don't think you need any of the header stuff in this test.

evan-m-jackson replied:
Agreed.

paytonrules commented on HTTPServerProject.Tests/HTTPServerReadTests/BodyTests.cs:
```diff
@@ -0,0 +1,25 @@
+        var headers = header.GetHeaders();
+
+        var body = new Body(reader);
+        var expected = body.GetBody();
```
This is `actual` not `expected` - your expected value is the hard coded "quit".

evan-m-jackson replied:
Agreed.

paytonrules reviewed (changes requested):
The review isn't done - as I mentioned in chat I ran out of time this morning - but I had a few comments I wanted you to see right away.

paytonrules commented on HTTPServerProject.Tests/HTTPServerResponseTests/StatusCodeTests.cs:
```diff
@@ -0,0 +1,19 @@
+public class UnitTestsForWritingResponses
+{
+    [Theory]
+    [InlineData(200, "HTTP/1.1 200 OK")]
```
Neat!

evan-m-jackson replied:
Thanks 🤙🏽

paytonrules commented on HTTPServerProject/HTTPServerResponse/ResponseByPath.cs:
```diff
@@ -7,8 +7,8 @@
-namespace HTTPServerProject.Path
-{
+namespace HTTPServerProject.Path;
+
```
It's not part of this PR - but I noticed it looking at this one. RequestPath doesn't need a default and probably shouldn't use one. You're always injecting a dictionary - which is a good thing! - so you probably don't want the user to be able to create a RequestPath without a dictionary. It would be empty, and essentially useless.

evan-m-jackson replied:
Agreed.

evan-m-jackson committed:
Add ToDo First ID capabilities and format

- Refactor initialized variables to begin with '_'
- Remove unneccessary usings in each class
- Rename Proxy test files
- Add First ID class to return first todo item's primary id
- Remove empty whitespaces

evan-m-jackson committed:
Add ToDo First ID capabilities and format

- Refactor initialized variables to begin with '_'
- Remove unneccessary usings in each class
- Rename Proxy test files
- Add First ID class to return first todo item's primary id
- Remove empty whitespaces

evan-m-jackson committed:
Remove dictionay paramter default in ResponsePath class

evan-m-jackson committed:
Remove dictionay paramter default in ResponsePath class

evan-m-jackson committed:
Refactor FirstID class

- Create RunToListRequest method that runs the WriteToDoListRequest method until the body string reads the correct string, which is the JSON object with the to-do ids

evan-m-jackson committed:
Refactor FirstID class

- Create RunToListRequest method that runs the WriteToDoListRequest method until the body string reads the correct string, which is the JSON object with the to-do ids

evan-m-jackson committed:
Refactor Proxy Response Class

- Create 'GetContentLength' method that returns the length of the response body from the Proxy Server request and implemented this method into the 'GetResponse' method.
- Create 'StatusWrapper' method that returns the status code based on the HTTP method. This allowed for the 'GetResponse' method's design to be cleaned up.

evan-m-jackson committed:
Refactor Proxy Response Class

- Create 'GetContentLength' method that returns the length of the response body from the Proxy Server request and implemented this method into the 'GetResponse' method.
- Create 'StatusWrapper' method that returns the status code based on the HTTP method. This allowed for the 'GetResponse' method's design to be cleaned up.

evan-m-jackson committed:
Remove unnecessary print statements

evan-m-jackson committed:
Remove unnecessary print statements

evan-m-jackson committed:
Merge pull request #17 from evan-m-jackson/proxy-server

Build Proxy Server Capabilities

evan-m-jackson committed:
Update README

- Add instructions for acceptance tests

evan-m-jackson committed:
Update README

- Add instructions for acceptance tests

evan-m-jackson committed:
Merge branch 'refactor-format' of https://github.com/evan-m-jackson/HTTP-Server into refactor-format

evan-m-jackson committed:
Add New Feature and Refactor Proxy Server

- Create UpdateInitialLine class to handle updating path for the proxy server request's
- Create interface called IGetFirstID to double GetFirstID (formerly FirstID) into TestGetFirstID class for testing UpdateInitialLine class
- Create Port class to allow the user to choose the port they want to run the server on. Made it default to port 5000.
- Light formatting and removing whitespaces

evan-m-jackson committed:
Add New Feature and Refactor Proxy Server

- Create UpdateInitialLine class to handle updating path for the proxy server request's
- Create interface called IGetFirstID to double GetFirstID (formerly FirstID) into TestGetFirstID class for testing UpdateInitialLine class
- Create Port class to allow the user to choose the port they want to run the server on. Made it default to port 5000.
- Light formatting and removing whitespaces

evan-m-jackson committed:
Refactor Port class

- Create IsPortValid method to return a boolean if the port number is between the range of 0 to 65535.
- Create GetPortNum method to returns the port string as an int or 5000 as default

evan-m-jackson committed:
Refactor Port class

- Create IsPortValid method to return a boolean if the port number is between the range of 0 to 65535.
- Create GetPortNum method to returns the port string as an int or 5000 as default

evan-m-jackson committed:
Refactor Github workflow files

evan-m-jackson committed:
Add acceptance test commands to Test YAML file

evan-m-jackson committed:
Add acceptance test commands to Test YAML file

evan-m-jackson committed:
Merge branch 'update-features' of https://github.com/evan-m-jackson/HTTP-Server into update-features

evan-m-jackson committed:
Separate acceptance test step into different steps

evan-m-jackson committed:
Test YAML

evan-m-jackson committed:
Update YAML file

evan-m-jackson committed:
Update Ruby uses step in YAML file

evan-m-jackson committed:
Format code and update Test YAML file

- Add named parameters to functions with more than 2 parameters.
- Add multiple lines for Lists with more than 2 items
- Remove acceptance tests for proxy server in Test YAML file
- Fix indentation

evan-m-jackson committed:
Test YAML file

evan-m-jackson committed:
Update Rakefile to return error code for tests

evan-m-jackson committed:
Add separate rake task for Acceptance Tests in Rakefile

evan-m-jackson committed:
Refactor 'Conversation' method

- Refactor 'Conversation' to have port number as parameter, set default to 5000.
- Refactor 'PathParameters' class to have port number as parameter in order to make '/redirect' path more flexible
= Refactor 'TestPathParameters' class to have default port number of 5000 to make tests pass

evan-m-jackson committed:
Refactor Test YAML file

- Shutdown server in 'Test' step

evan-m-jackson committed:
Merge pull request #18 from evan-m-jackson/update-features

Add New Feature and Refactor Proxy Server

evan-m-jackson committed:
Refactor Test YAML

paytonrules commented:
With 48 commits you're gonna want so squash these into a clearer history. Probably not just one, but a few that explain what you did. (For instance you have two commits that say "Light Refactor of Code" - that could be one commit with a better title of what you did to refactor the code.)

paytonrules commented:
There's a ton here now - let's get a better title and description on the PR.

paytonrules commented on HTTPServerProject.Tests/HTTPServerWriteTests/TestWriteStreams.cs:
```diff
@@ -0,0 +1,46 @@
+            }
+        }
+
+        public void Flush()
```
I'm surprised these don't have booleans set in them - you could use these to test if something is flushed and closed.

evan-m-jackson replied:
Agreed.

paytonrules commented on HTTPServerProject.Tests/HTTPServerProxyTests/TestClient.cs:
```diff
@@ -0,0 +1,16 @@
+public class TestProxyClient : IProxyClient
+{
+    List<string> stream = new List<string>();
+    public TestProxyClient()
```
I don't believe the empty constructor is necessary.

evan-m-jackson replied:
Agreed.

paytonrules commented on HTTPServerProject.Tests/HTTPServerProxyTests/TestClient.cs:
```diff
@@ -0,0 +1,16 @@
+
+namespace HTTPServerProxyTests.Client;
+
+public class TestProxyClient : IProxyClient
```
See my broader note attached to the bottom about `IProxyClient`

evan-m-jackson replied:
I am going to remove the empty interface `IProxyClient` and this unneccessary test double

paytonrules commented on HTTPServerProject.Tests/HTTPServerProxyTests/TestFirstID.cs:
```diff
@@ -0,0 +1,103 @@
+
+namespace HTTPServerProxyTests.FirstID;
+
+public class TestGetFirstID : IGetFirstID
```
What purpose does this serve? It looks like a complete copy of `GetFirstID`.

evan-m-jackson replied:
I  changed the `WriteToDoListRequest` method in the `TestGetFirstID` to make a test double.
